### PR TITLE
docs: adds links and support statement for managed offerings

### LINKED ISSUE #1095: update support to clarify aks addon and arc extensions are supported
**Describe the solution you'd like**
https://github.com/Azure/secrets-store-csi-driver-provider-azure#support

we should add links to the Arc extension and the AKS addon and clarify those are supported by each platform's support model

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Secrets Store CSI Driver version: (use the image tag):
- Azure Key Vault provider version: (use the image tag):
- Kubernetes version: (use `kubectl version`):
- Cluster type: (e.g. AKS, aks-engine, etc):

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**: 
Fixes  #1095 

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

**Comment by nilekhc** [comment]
cc @ritazh

**Review by sozercan — APPROVED** [review]

**Review by aramase — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -43,3 +43,5 @@ This project has adopted the [Microsoft Open Source Code of Conduct](https://ope
 ## Support
 
 Azure Key Vault Provider for Secrets Store CSI Driver is an open source project that is [**not** covered by the Microsoft Azure support policy](https://support.microsoft.com/en-us/help/2941892/support-for-linux-and-open-source-technology-in-azure). [Please search open issues here](https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues), and if your issue isn't already represented please [open a new one](https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/new/choose). The project maintainers will respond to the best of their abilities.
+
+However, [Arc extension](https://learn.microsoft.com/en-us/azure/azure-arc/kubernetes/tutorial-akv-secrets-provider) and [AKS addon](https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver) for _Azure Key Vault Provider for Secrets Store CSI Driver_ are managed offerings and supported by Microsoft Azure support policy.
```
Comment (ritazh) [inline README.md:47]: Maybe I missed it but I could not find a link to the AKS addon or Arc extension in the website or readme. Can we update the landing page so it is very easy to discover them?

Comment (nilekhc) [reply README.md:47]: This is the main Readme. Please let me know if there is a better way to surface this here.

I have also added it to the main landing page of the website. https://deploy-preview-1094--secrets-store-csi-driver-provider-azure.netlify.app/#managed-add-ons

**Review by ritazh — APPROVED** [review]
LGTM

**Pull request merged**